### PR TITLE
Frontend/user page bugs

### PR DESCRIPTION
**Pull request by kittybest**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Build (`yarn build`) was run locally and any changes were pushed
- [ ] Lint (`yarn lint --check`) has passed locally and any fixes were made for failures

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

-
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Comment by kittybest** [comment]
Remind: Needs to run `yarn build` first, since the files in `core` have been changed.

File: packages/core/src/schema.ts [hunk 1]
```diff
@@ -109,21 +109,6 @@ const _schema = [
             },
         ],
     },
-    {
-        name: 'EpkRecord',
```
Comment (AnthonyMadia) [inline packages/core/src/schema.ts:113]: Changes to the schema may require the docker image to be re-bulit cc: @vimwitch

File: packages/frontend/src/context/User.ts [hunk 2]
```diff
@@ -106,6 +110,59 @@ export class User {
...
+                from: BigInt(r.from)
+                    .toString(16)
+                    .padStart(this.unirepConfig.epochTreeDepth / 4, '0'),
+                to: BigInt(r.to)
+                    .toString(16)
+                    .padStart(this.unirepConfig.epochTreeDepth / 4, '0'),
+            }
+        })
+
+        this.recordsByEpk = {} // maybe it's not a good solution
```
Comment (AnthonyMadia) [inline packages/frontend/src/context/User.ts:133]: Hard to follow what this code is doing considering there is already a state variable with the name `recordsByEpk`

File: packages/backend/test/utils.ts [hunk 3]
```diff
@@ -123,17 +123,18 @@ export const getSpent = async (t, iden) => {
...
     }
     const paramStr = epks.join('_')
-    const r = await fetch(
-        `${t.context.url}/api/records/${paramStr}?spentonly=true`
-    )
+    const r = await fetch(`${t.context.url}/api/records/${paramStr}`)
     const data = await r.json()
     if (!r.ok) {
         throw new Error(`/records error ${JSON.stringify(data)}`)
     }
```
Comment (AnthonyMadia) [inline packages/backend/test/utils.ts:130]: We should also probably do the error handling for the response right after we are awaiting the fetch.
So, it would be:
```js
const r = await fetch (..)
if (!r.ok) { .. throw the error}
const data = await r.json()
```
I think in the current form, we would not get the error handling we expect.

Comment (AnthonyMadia) [reply packages/backend/test/utils.ts:130]: The current form for error handling is also being used in other functions within this file.

**Comment by vivianjeng** [comment]
I found that the **boost** and **squash** field become **+0** and **-0** after refreshing the user info page

https://user-images.githubusercontent.com/17507085/205879565-db9e186e-4c4c-4399-994d-b46fe464306d.mov

File: packages/frontend/src/pages/userPage/userPage.tsx [hunk 4]
```diff
@@ -90,48 +91,63 @@ const UserPage = () => {
...
-                if (isReceived) {
-                    // right stuff
-                    if (h.action === ActionType.UST) {
-                        r[0] += h.upvote
-                    } else if (h.action === ActionType.Vote) {
-                        r[1] += h.upvote
-                        r[2] += h.downvote
-                    }
-                }
+        console.log(user.recordsByEpk)
```
Comment (vivianjeng) [inline packages/frontend/src/pages/userPage/userPage.tsx:98]: the debug info can be removed

File: packages/frontend/src/context/User.ts [hunk 5]
```diff
@@ -36,6 +38,7 @@ export class User {
             netReputation: computed,
             subsidyReputation: observable,
             spent: observable,
+            // recordsByEpk: observable,
```
Comment (vivianjeng) [inline packages/frontend/src/context/User.ts:41]: can be removed?

File: packages/backend/test/utils.ts [hunk 6]
```diff
@@ -212,6 +215,7 @@ export const createPost = async (t, iden) => {
 }
 
 export const createPostSubsidy = async (t, iden) => {
+    console.log('call create post subsidy')
```
Comment (vivianjeng) [inline packages/backend/test/utils.ts:218]: the debug info can be removed

File: packages/backend/test/utils.ts [hunk 7]
```diff
@@ -186,19 +189,19 @@ export const createPost = async (t, iden) => {
...
+
+    const data = await r.json()
     const receipt = await t.context.provider.waitForTransaction(
         data.transaction
     )
 
     for (;;) {
         await new Promise((r) => setTimeout(r, 1000))
         const currentSpent = await getSpent(t, iden)
+        console.log('current spent: ', currentSpent)
```
Comment (vivianjeng) [inline packages/backend/test/utils.ts:204]: the debug info can be removed

**Review by vivianjeng — APPROVED** [review]
LGTM

File: packages/frontend/src/context/User.ts (hunk 2 above)
Comment (kittybest) [reply packages/frontend/src/context/User.ts:133]: since it will reload the records from db each time this function is called, and if it wasn't cleared, the records are duplicated. ofc we can do things like if a record exists already then just don't do anything to it, but it costs time to do this every time, so i choose to just let it be empty again and add records into it.

**Comment by kittybest** [comment]
> I found that the **boost** and **squash** field become **+0** and **-0** after refreshing the user info page
> 
>  user-page-bug.mov

this should be an issue, since it exists also in `main`, https://github.com/Unirep/Unirep-Social/issues/178

File: packages/frontend/src/context/User.ts [hunk 8]
```diff
@@ -106,6 +109,59 @@ export class User {
...
 
+    async loadRecords() {
+        const epksBase10 = this.allEpks.map((epk) =>
+            BigInt('0x' + epk).toString()
+        )
+
+        const apiURL = makeURL(`records/${epksBase10.join('_')}`, {})
+        const r = await fetch(apiURL)
+        const data = await r.json()
+        const rawRecords = data.map((r: Record) => {
```
Comment (AnthonyMadia) [inline packages/frontend/src/context/User.ts:120]: I would also add error handing for if the fetch request fails here. 

For example:
`if (!r) {  .... handle error ... }`

Comment (kittybest) [reply packages/frontend/src/context/User.ts:120]: oooops, sorry I forgot, thx for finding this :)
